fix: updated README to follow latest steps for domain setup (#14)
WalletConnect/gm-hackers: README.md

https://www.loom.com/share/a7001711b8a94500b827a9d2655c8654?sid=1a36ccfa-9ee8-42a5-a882-9b78ac6e846a

## Running locally

Install the app's dependencies:

```bash
npm i
```

## Develop

```bash

> npm run dev

Reviewer: This section ignores that it depends on Expose domain below. There should be a mention of this dependency so people don't get lost.

```

## Build

```bash
npm run build
```

## Expose domain

> The `did.json` file needs to be hosted on a publicly available domain. Although Vercel and the like work great for the actual deployment, any dev working with notify might like a hot-reloadable deployment. This is where tunnels come in.

Reviewer: "and the like" might not be a phrase generally understood by non-English speakers

> The `did.json` file needs to be hosted on a publicly available domain. Although Vercel and the like work great for the actual deployment, any dev working with notify might like a hot-reloadable deployment. This is where tunnels come in.

Reviewer: ```suggestion
The `did.json` file needs to be hosted on a publicly available domain. Although Vercel and the like work great for the actual deployment, any dev working with Notify API might like a hot-reloadable deployment. This is where tunnels come in.
```

> The `did.json` file needs to be hosted on a publicly available domain. Although Vercel and the like work great for the actual deployment, any dev working with notify might like a hot-reloadable deployment. This is where tunnels come in.

Reviewer: > hot-reloadable deployment

Misleading. If the did.json is deployed to a non-hot-reloadable location (e.g. Vercel), they can still do the rest of their development locally in a hot-reloadable deployment.

Follow the following instructions to expose your app from localhost to be publicly available:

[Instructions adapted from Cloudflare](https://developers.cloudflare.com/cloudflare-one/connections/connect-networks/do-more-with-tunnels/trycloudflare/#use-trycloudflare)

1. Download `cloudflared` utility:
   - MacOS: `brew install cloudflared`
   - Ubuntu/Debian: `apt install cloudflared`
   - Windows: [Download from here](https://developers.cloudflare.com/cloudflare-one/connections/connect-networks/downloads/)
2. Run your app in localhost

```sh
npm run dev
```

3. Expose your app

```sh

> cloudflared tunnel --url http://localhost:3000

Reviewer: Probably should mention that this domain is ephemeral so if the tunnel is restarted you will get a new domain. And you will need to update that new domain in Cloud App. The tunnel needs to continue running for the duration of any tests with Notify API.

For deploying the app to a public URL, refer to the [Deploy](#deploy-the-example-dapp) section.

```bash
npm run build
```

## Expose domain

The `did.json` file needs to be hosted on a publicly available domain. Although Vercel and similar alternatives work great for the actual deployment, any dev working with Notify API might like a hot-reloadable deployment. This is where tunnels come in.
Follow the following instructions to expose your app from localhost to be publicly available:

[Instructions adapted from Cloudflare](https://developers.cloudflare.com/cloudflare-one/connections/connect-networks/do-more-with-tunnels/trycloudflare/#use-trycloudflare)

1. Download `cloudflared` utility:
   - MacOS: `brew install cloudflared`
   - Ubuntu/Debian: `apt install cloudflared`
   - Windows: [Download from here](https://developers.cloudflare.com/cloudflare-one/connections/connect-networks/downloads/)
2. Run your app in localhost

```sh
npm run dev
```

3. Expose your app

```sh
cloudflared tunnel --url http://localhost:3000
```

Once you've got that running, you can access your local app from a public domain that looks like:

```
https://some-combination-of-words.trycloudflare.com
```

> Make sure this tunnel is running in the background while you are testing your app. If the tunnel is closed, you will need to update the dapp url and use the updated `did.json` file from [WalletConnect Cloud](https://cloud.walletconnect.com).

Reviewer: Actually they don't need to update the did.json since it is still the same project ID and doesn't change.

```suggestion
Make sure this tunnel is running in the background while you are testing your app. If the tunnel is closed, you will need to update the dapp URL in [WalletConnect Cloud](https://cloud.walletconnect.com).
```

Once you've got that running, you can now use this domain and similar configuration to test your app with notify:

1. Head over to [WalletConnect Cloud](https://cloud.walletconnect.com) and Sign in or Sign up if you don't have an account.
2. Create a project and take note of your Project ID.
3. You will need to set the `NEXT_PUBLIC_PROJECT_ID` environment variable to your Project ID from step #2.
4. Back in WalletConnect Cloud, navigate to your project's APIs tab. Under Notify API > Configuration > `DAPP INFORMATION`, provide your cloudflare URL as the dapp URL.
5. Under the same section, next to Notification types, click on the "Add Notification Type" button and add a title and description for your notification type. This is the type of notification that your app will publish.
   For example, if you are going to send promotional content as notification, you might want to add a notification type called "Promotional" with a description "Promotional content from the XYZ Team.".

>    Click the copy button next to your newly created notification type to copy its ID. Replace the existing ID in `/pages/index.tsx` Line 120 with your new ID.

Reviewer: Need to save first to get the ID.

```suggestion
   After saving, click the copy button next to your newly created notification type to copy its ID. Replace the existing ID in `/pages/index.tsx` Line 120 with your new ID.
```

   - [Create repo & Deploy to Vercel](https://vercel.com/new/clone?repository-url=https%3A%2F%2Fgithub.com%2FWalletConnect%2Fgm-hackers&env=NEXT_PUBLIC_PROJECT_ID&envDescription=Get%20your%20Project%20ID%20on%20WalletConnect%20Cloud.&envLink=https%3A%2F%2Fcloud.walletconnect.com%2F)
   - [Create repo & Deploy to Netlify](https://app.netlify.com/start/deploy?repository=https://github.com/WalletConnect/gm-hackers)
   - Fork/clone this repo and deploy yourself
4. Back in the WalletConnect Cloud, navigate to your project's APIs tab. Under Notify API > Configuration > `DAPP INFORMATION`, provide your public URL as the dapp URL.
5. Under the same section, next to Notification types, click on the "Add Notification Type" button and add a title and description for your notification type. This is the type of notification that your app will publish.

>    For example, if you are going to send promotional content as notification, you might want to add a notification type called "Promotional" with a description "Promotional content from the XYZ Team.".

Reviewer: They also need to copy the notification type ID from this, which only appears after saving. They need this for the type field when sending the notification

6. Still on Notify API section, you should see a `Notify API Secret`. Copy this secret into your deployment as the `NOTIFY_API_SECRET` environment variable. Make sure to update the environment variables on your local environment as well as on your deployment platform.

> 7. Next, you will need to host `did.json` file on this page at the `/.well-known/` directory of your public URL. You can do this by saving them to the `/public/.well-known/` directory of your fork of this template repo. Note that you will need to overwrite the existing `did.json` file.

Reviewer 2: ```suggestion
7. Next, you need to host the `did.json` file on your public URL in the /.well-known/ directory. Save it to the /public/.well-known/ directory in your fork of this template repository. Be sure to replace the existing did.json file.
```
